content/blog/gitlab-ci-component.md

```diff
@@ -0,0 +1,164 @@
+      keycloak_client_id: $KEYCLOAK_CLIENT_ID
+      keycloak_client_secret: $KEYCLOAK_CLIENT_SECRET
+      stage: import
+
+stages:
+  - import
+```
+
+Some important things to notice here:
+* You can customize the `stage` name to fit your pipeline structure
+* Using `@~latest` ensures you always get the most recent component version
+
+Once this runs, your API specifications will be imported into Microcks, and your development team can immediately start consuming mock endpoints!
+
+## Running Contract Conformance Tests
+
+The second component, `microcks-test`, runs contract tests against your deployed API endpoints. This ensures your implementation matches your specification—automatically catching breaking changes before they reach production!
+
+Here's how to use it:
+
+```yaml
+# .gitlab-ci.yml
+include:
+  - component: gitlab.com/microcks-cncf/microcks-community/microcks-gitlab-components/microcks-test@~latest
+    inputs:
+      api_name_version: "Weather Forecast API:1.0.0"
+      test_endpoint: "https://my-weather-api.example.com"
+      test_runner: "OPEN_API_SCHEMA"
+      microcks_url: "https://microcks.apps.acme.com/api/"
+      keycloak_client_id: $KEYCLOAK_CLIENT_ID
+      keycloak_client_secret: $KEYCLOAK_CLIENT_SECRET
+      stage: test
+      wait_for: "15sec"
+
+stages:
+  - test
+```
+
+## Putting It All Together: A Pipeline sample
+
+Let's see how both components work together in a real-world pipeline. [The following example](https://gitlab.com/saberkan_personal/microcks-gitlab-components/-/tree/testing_components?ref_type=heads) demonstrates how to import a local spec file, and how to run the test against a running API.
+
+To run this example, you will need to:
+- Configure `$MICROCKS_DN` and `$PASTRY_API_DN` as GitLab CI variable.
+- Run the [Pastry API](https://microcks.io/documentation/tutorials/getting-started-tests/#deploying-the-api-implementation)
+- Configure a running [Microcks instance in Dev mode](https://microcks.io/documentation/guides/installation/docker-compose/#development-mode)
+- Load the [Pastry sample](https://microcks.io/documentation/tutorials/getting-started/#loading-a-sample) 
+
+
+```yaml
+# .gitlab-ci.yml
+include:
+  - component: $CI_SERVER_FQDN/microcks-cncf/microcks-community/microcks-gitlab-components/microcks-import@~latest
+    inputs:
+      specs: "samples/HelloService.postman.json:true"
+      microcks_url: "http://$MICROCKS_DN:8080/api/"
+      stage: import
+  - component: $CI_SERVER_FQDN/microcks-cncf/microcks-community/microcks-gitlab-components/microcks-test@~latest 
+    inputs:
+      api_name_version: "API Pastry - 2.0:2.0.0"
+      test_endpoint: "http://$PASTRY_API_DN/"
+      test_runner: "OPEN_API_SCHEMA"
+      microcks_url: "http://$MICROCKS_DN:8080/api/"
+      stage: test
+
+stages:
+  - import
+  - test
+```
+
+If the contract test fails, the pipeline stops, preventing broken APIs from reaching production. It's continuous verification in action!
+
```

Comment:
I wonder if a screenshot of the "all together" pipeline wouldn't be a nice addition?